packages/eslint-plugin/tests/rules/prefer-find.test.ts

```diff
@@ -584,5 +590,142 @@ arr.find(f, thisArg);
         },
       ],
     },
+    {
+      code: `
+(Math.random() < 0.5
+  ? [1, 2, 3].filter(x => false)
+  : [1, 2, 3].filter(x => true))[0];
+      `,
+      errors: [
+        {
+          line: 2,
+          messageId: 'preferFind',
+          suggestions: [
+            {
+              messageId: 'preferFindSuggestion',
+              output: `
+(Math.random() < 0.5
+  ? [1, 2, 3].find(x => false)
+  : [1, 2, 3].find(x => true));
+      `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+Math.random() < 0.5
+  ? [1, 2, 3].find(x => true)
+  : [1, 2, 3].filter(x => true)[0];
+      `,
+      errors: [
+        {
+          line: 4,
+          messageId: 'preferFind',
+          suggestions: [
+            {
+              messageId: 'preferFindSuggestion',
+              output: `
+Math.random() < 0.5
+  ? [1, 2, 3].find(x => true)
+  : [1, 2, 3].find(x => true);
+      `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+declare const f: (arg0: unknown, arg1: number, arg2: Array<unknown>) => boolean,
+  g: (arg0: unknown) => boolean;
+const nestedTernaries = (
+  Math.random() < 0.5
+    ? Math.random() < 0.5
+      ? [1, 2, 3].filter(f)
+      : []?.filter(x => 'shrug')
+    : [2, 3, 4]['filter'](g)
+).at(0.2);
+      `,
+      errors: [
+        {
+          line: 4,
+          messageId: 'preferFind',
+          suggestions: [
+            {
+              messageId: 'preferFindSuggestion',
+              output: `
+declare const f: (arg0: unknown, arg1: number, arg2: Array<unknown>) => boolean,
+  g: (arg0: unknown) => boolean;
+const nestedTernaries = (
+  Math.random() < 0.5
+    ? Math.random() < 0.5
+      ? [1, 2, 3].find(f)
+      : []?.find(x => 'shrug')
+    : [2, 3, 4]["find"](g)
+);
+      `,
+            },
+          ],
+        },
+      ],
+    },
+
+    {
+      code: `
+declare const f: (arg0: unknown) => boolean, g: (arg0: unknown) => boolean;
+const nestedTernariesWithSequenceExpression = (
+  Math.random() < 0.5
+    ? ('sequence',
+      'expression',
+      Math.random() < 0.5 ? [1, 2, 3].filter(f) : []?.filter(x => 'shrug'))
+    : [2, 3, 4]['filter'](g)
+).at(0.2);
+      `,
+      errors: [
+        {
+          line: 3,
+          messageId: 'preferFind',
+          suggestions: [
+            {
+              messageId: 'preferFindSuggestion',
+              output: `
+declare const f: (arg0: unknown) => boolean, g: (arg0: unknown) => boolean;
+const nestedTernariesWithSequenceExpression = (
+  Math.random() < 0.5
+    ? ('sequence',
+      'expression',
+      Math.random() < 0.5 ? [1, 2, 3].find(f) : []?.find(x => 'shrug'))
+    : [2, 3, 4]["find"](g)
+);
+      `,
+            },
+          ],
+        },
+      ],
+    },
+
+    {
+      code: `
+declare const spreadArgs: [(x: unknown) => boolean];
+[1, 2, 3].filter(...spreadArgs).at(0);
```

Comment:
not related, just an extra test I wish I'd written on the initial implementation